### ISSUE TITLE
Add class and _include parameter to selectively show some aside blocks in smallest viewports?

### ISSUE THREAD

**Issue by alexgreen** [TheTyee/design-article.thetyee.ca#83]

**Comment by alexgreen** [comment]
To mitigate against crowding on smaller phone screens I've set all aside blocks to display:none in viewports narrower than 500px. 

This means some content won't be visible/available to readers at this smallest size. However, most of this content is soft promotion (donate aside, related articles, related events) so these things probably shouldn't be shown on tiny screens anyway.

However there's a couple kinds of aside blocks that are more closely related to the article content: aside containing images, asides containing a series box. (Anything else?) And these are gone too. Is this a problem?

For annoying CSS reasons the visibility of aside blocks has to be controlled at the <aside> container level, not at the level of the contained block. So targeting specific kinds of asides is tricky. However I could add an extra class the <aside> level to maintain visibility for some kinds of asides. (But this is a bit messy.)

...or maybe we could use the polyfill to implement the new :has CSS4 pseudo class and style asides based on their contents (dream come true, but maybe overkill):
https://drafts.csswg.org/selectors-4/#relational
http://www.ericponto.com/blog/2015/01/10/has-pseudo-class-parent-selector/

**Comment by phillipadsmith** [comment]
> This means some content won't be visible/available to readers at this smallest size. However, most of this content is soft promotion (donate aside, related articles, related events) so these things probably shouldn't be shown on tiny screens anyway.

Hmmm... we should run this by Tyee staff before making that decision. 

> However there's a couple kinds of aside blocks that are more closely related to the article content: aside containing images, asides containing a series box. (Anything else?) And these are gone too. Is this a problem?

Really a question for Robyn, as those are both editorial concerns. Can you raise this in Basecamp, @alexgreen, with a screenshot showing the difference? 

The argument can probably be made that "less is more" on screens that small, and yet there are also those arguments in favour of not "hiding" content from mobile visitors... so we'll have to ask Robyn if those asides constitute content or not.

I agree, however, that the view is much nicer without the asides. 

Phillip.

**Comment by kidcompassion** [comment]
Not sure if this was raised with editorial, but I put it here just in case: https://basecamp.com/1929296/projects/9481270/messages/56891501

**Comment by alexgreen** [comment]
5432a0b77a08d150341cb0ceda3c1c2209b462f3
I implemented the polyfill to implement the new :has CSS4 pseudo class and style asides based on their contents:
https://drafts.csswg.org/selectors-4/#relational
http://www.ericponto.com/blog/2015/01/10/has-pseudo-class-parent-selector/

The idea is that we can now select for asides that contain info essential to the story and display only those in the smallest viewports while hiding all asides with tertiary content (such as promotional messaging that also gets displayed below the article__body).

Currently I've set it to display asides with img children. This is reflected on the library page and the standard-article page. If other asides need to be displayed then we can easily add them.
